pkg/vmprovider/providers/vsphere2/constants/constants.go

```diff
@@ -134,4 +134,17 @@ const (
 	V1alpha2SubnetMask = "V1alpha2_SubnetMask"
 	// V1alpha2FormatNameservers is an alias for versioned templating function V1alpha2_FormatNameservers.
 	V1alpha2FormatNameservers = "V1alpha2_FormatNameservers"
+
+	// BackupVMKubeDataExtraConfigKey is the ExtraConfig key to the VirtualMachine
+	// resource's Kubernetes spec data, compressed using gzip and base64-encoded.
+	BackupVMKubeDataExtraConfigKey = "vmservice.virtualmachine.kubedata"
+	// BackupVMBootstrapDataExtraConfigKey is the ExtraConfig key to the VM's
+	// bootstrap data object, compressed using gzip and base64-encoded.
+	BackupVMBootstrapDataExtraConfigKey = "vmservice.virtualmachine.bootstrapdata"
```

Comment:
For v1a2 vApp customization there is also be a map of maps: see GetVirtualMachineBootstrap(). I'm pretty sure - once we get around to supporting it - the inlined CloudInit customization can also specify a map of maps.

I'm fine if we just what to worry about this case later.

Reply:
Yes, I have also left a todo comment to address this case. I will work on it once the v1a2 customization is ready (thought it would be easier to validate my change then).